index.js

```diff
@@ -7,440 +8,217 @@
+                let series = results[0];
+                series.Episodes = results[1];
+                return series;
+            });
+    }
 
-                return done(error);
-            } else if (error) {
-                return done(error);
+    /**
+     * @private
+     */
+    login() {
+        this.loginPromise = request.post({
+            baseUrl,
+
+            json: true,
+            uri: 'login',
+            body: {
+                apikey: this.apiKey
             }
-
-            if (responseType === RESPONSE_TYPE.ZIP) {
-
-                Zip.loadAsync(data).then(function(zip) {
-                    return zip.file(`${urlOpts.lang}.xml`).async("string")
-                }).then(function(contents) {
-                    parseXML(contents, normalise, done);
-                }).catch(function(error) {
-                    done(error);
+        })
+            .then(data => data.token)
+            .then(token => {
+                this.defaultRequest = request.defaults({
+                    baseUrl,
+
+                    json: true,
+                    headers: {
+                        "User-Agent": "edwellbrook/node-tvdb",
+                        Authorization: `Bearer ${token}`,
+                        'Accept-language': this.language
+                    }
                 });
+            });
+    }
 
-            } else {
-                parseXML(data, normalise, done);
-            }
-        });
-    });
-}
-
-/**
- * Parse XML response
- *
- * @param {String} xml data
- * @param {Function} normalise - a function to tidy the response object
- * @param {Function} callback
- * @api private
- */
+    /**
+     * @private
+     */
+    requestGet(options, language) {
+        options = defaults({}, options, {headers: {'Accept-language': language}});
+        return this.loginPromise
+            .then(() => this.defaultRequest.get(options))
+            .then(response => {
+                if (hasNextPage(response)) {
+                    return this.getNextPage(response, options)
+                        .then(nextPageResponse => [response.data, nextPageResponse])
+                        .then(dataArray => flatten(dataArray));
+                }
+                return response.data;
+            });
+    }
 
-function parseXML(data, normalise, callback) {
-    parser(data, PARSER_OPTS, function(error, results) {
-        if (results && results.Error) {
-            return callback(new Error(results.Error));
-        }
+    /**
+     * @private
+     */
+    getNextPage(response, options) {
+        let separator = options.uri.indexOf('?') === -1 ? '?' : '&';
+        let uri = options.uri + separator + 'page=' + response.links.next;
+        let o = defaults({uri}, options);
+        return this.requestGet(o);
+    }
 
-        normalise(results, function(results) {
-            callback(error, results);
-        });
-    });
+    /**
+     * @param {String} url
+     */
+    static setBaseUrl(url) {
```

Comment:
Not convinced this is needed right now. Can we remove this method and initialise the base URL on line 15 with a `const` thanks.